### PR TITLE
HV-536 Render placholders inside a new placholder-container

### PR DESCRIPTION
**Pull request by lancefisher-wf**
Problem
================

In flow mode, rendering placeholder elements as direct children of the transformation plane causes problems when other users of the transformation plane inject content as well. Since we remove and append these placeholders as a user scrolls through a document, it can cause any other content that relies on being ordered after the placeholders to not work.

Solution
====================

Render the placeholder elements in a new placeholder-container on the transformation plane. Other users can safely inject content before or after the placeholder-container, and not worry about it being mixed up with placeholders.

Testing
=======================

Run the example scroll list app, and test each scroll mode. Each scroll mode should continue to work, and placeholders should be added and removed as you scroll through the document.

Additionally, test this in your own application, @todbachman-wf.

### CODE REVIEW

**Comment by rmconsole-wf** [comment]
When this pull is merged I will not be able to function
		properly because no ticket is included in the title.
		If your team has decided not to use jira please indicate this in rmconsole

***
*Last updated on Tuesday, April 07 05:46 PM CST*

**Comment by lancefisher-wf** [comment]
@rmconsole-wf yo

File: src/scroll_list/PlaceholderRenderer.js [hunk 1]
```diff
@@ -459,6 +470,28 @@ define(function(require) {
...
             }
+        },
+
+        getPlaceholderContainer: function() {
+            if (this._placeholderContainer) {
+                return this._placeholderContainer;
+            }
+
+
+            var scrollList = this._scrollList;
```
Comment (todbachman-wf) [inline src/scroll_list/PlaceholderRenderer.js]: We could tighten up the whitespace above.

File: src/scroll_list/PlaceholderRenderer.js [hunk 2]
```diff
@@ -459,6 +470,28 @@ define(function(require) {
...
+            var listMap = scrollList.getListMap();
+            var transformationPlane = listMap.getTransformationPlane();
+            var placeholderContainer = transformationPlane.querySelector('.placeholder-container');
+            if (placeholderContainer) {
+                return placeholderContainer;
+            }
```
Comment (todbachman-wf) [inline src/scroll_list/PlaceholderRenderer.js]: We should add `this._placeholderContainer = placeholderContainer` to the block above.

**Comment by todbachman-wf** [comment]
+1. Just some minor comments. This is going to be so much better!

File: src/scroll_list/PlaceholderRenderer.js [hunk 3]
```diff
@@ -459,6 +470,28 @@ define(function(require) {
...
+
+            placeholderContainer = document.createElement('div');
+            placeholderContainer.className = 'placeholder-container';
```
Comment (patkujawa-wf) [inline src/scroll_list/PlaceholderRenderer.js]: Does this align with other awesomemap class names? Don't they have awesomeMap as a prefix?

**Comment by patkujawa-wf** [comment]
+1 as well, in case @todbachman-wf wants to p10

File: src/scroll_list/PlaceholderRenderer.js (hunk 2 above)
Comment (lancefisher-wf) [inline src/scroll_list/PlaceholderRenderer.js]: Ah right, I meant to do that. I was testing without caching...

**Comment by todbachman-wf** [comment]
+10.

File: src/scroll_list/PlaceholderRenderer.js (hunk 1 above)
Comment (lancefisher-wf) [inline src/scroll_list/PlaceholderRenderer.js]: Fixed.

File: src/scroll_list/PlaceholderRenderer.js (hunk 3 above)
Comment (lancefisher-wf) [inline src/scroll_list/PlaceholderRenderer.js]: Renamed to `scrollList-placeholderContainer`. The AwesomeMap is unaware of what we're doing here :)

**Comment by todbachman-wf** [comment]
+10 again

**Comment by patkujawa-wf** [comment]
+1 again

**Comment by charliekump-wf** [comment]
QA Resource Approval: +1

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Unit test created/updated 
- [x] All unit tests pass 

Merging into master.

**Pull request merged**